// @aws-sdk/client-lambda/dist/es/commands/CreateCodeSigningConfigCommand.js
import { __extends } from "tslib";
import { CreateCodeSigningConfigRequest, CreateCodeSigningConfigResponse } from "../models/models_0";
import { deserializeAws_restJson1CreateCodeSigningConfigCommand, serializeAws_restJson1CreateCodeSigningConfigCommand, } from "../protocols/Aws_restJson1";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
/**
 * <p>Creates a code signing configuration. A <a href="https://docs.aws.amazon.com/lambda/latest/dg/configuration-trustedcode.html">code signing configuration</a> defines a list of
 *       allowed signing profiles and defines the code-signing validation policy (action to be taken if deployment
 *       validation checks fail). </p>
 */
var CreateCodeSigningConfigCommand = /** @class */ (function (_super) {
    __extends(CreateCodeSigningConfigCommand, _super);
    // Start section: command_properties
    // End section: command_properties
    function CreateCodeSigningConfigCommand(input) {
        var _this = 
        // Start section: command_constructor
        _super.call(this) || this;
        _this.input = input;
        return _this;
        // End section: command_constructor
    }
    /**
     * @internal
     */
    CreateCodeSigningConfigCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "LambdaClient";
        var commandName = "CreateCodeSigningConfigCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: CreateCodeSigningConfigRequest.filterSensitiveLog,
            outputFilterSensitiveLog: CreateCodeSigningConfigResponse.filterSensitiveLog,
        };
        if (typeof logger.info === "function") {
            logger.info({
                clientName: clientName,
                commandName: commandName,
            });
        }
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    CreateCodeSigningConfigCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1CreateCodeSigningConfigCommand(input, context);
    };
    CreateCodeSigningConfigCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1CreateCodeSigningConfigCommand(output, context);
    };
    return CreateCodeSigningConfigCommand;
}($Command));
export { CreateCodeSigningConfigCommand };
