// @aws-sdk/client-lambda/dist/es/commands/ListFunctionsCommand.js
import { __extends } from "tslib";
import { ListFunctionsRequest, ListFunctionsResponse } from "../models/models_0";
import { deserializeAws_restJson1ListFunctionsCommand, serializeAws_restJson1ListFunctionsCommand, } from "../protocols/Aws_restJson1";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
/**
 * <p>Returns a list of Lambda functions, with the version-specific configuration of each. Lambda returns up to 50
 *       functions per call.</p>
 *          <p>Set <code>FunctionVersion</code> to <code>ALL</code> to include all published versions of each function in
 *       addition to the unpublished version. To get more information about a function or version, use <a>GetFunction</a>.</p>
 */
var ListFunctionsCommand = /** @class */ (function (_super) {
    __extends(ListFunctionsCommand, _super);
    // Start section: command_properties
    // End section: command_properties
    function ListFunctionsCommand(input) {
        var _this = 
        // Start section: command_constructor
        _super.call(this) || this;
        _this.input = input;
        return _this;
        // End section: command_constructor
    }
    /**
     * @internal
     */
    ListFunctionsCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "LambdaClient";
        var commandName = "ListFunctionsCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: ListFunctionsRequest.filterSensitiveLog,
            outputFilterSensitiveLog: ListFunctionsResponse.filterSensitiveLog,
        };
        if (typeof logger.info === "function") {
            logger.info({
                clientName: clientName,
                commandName: commandName,
            });
        }
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    ListFunctionsCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1ListFunctionsCommand(input, context);
    };
    ListFunctionsCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1ListFunctionsCommand(output, context);
    };
    return ListFunctionsCommand;
}($Command));
export { ListFunctionsCommand };
